test/test_user_settings.py

```diff
@@ -44,11 +49,21 @@ def setUpClass(self):
     os.environ["NOT_PARSED"] = "ignored"
 
 
-
   @classmethod
   def tearDownClass(self):
     os.chdir(self.working_dir)
 
+    # Other unittests might depend on defaults:
+    # Restore monkey patched settings ...
+    for key, val in self.settings_backup.iteritems():
+      setattr(in_toto.settings, key, val)
+
+    # ... and delete test environment variables
+    del os.environ["IN_TOTO_ARTIFACT_EXCLUDE_PATTERNS"]
+    del os.environ["IN_TOTO_ARTIFACT_BASE_PATH"]
+    del os.environ["IN_TOTO_not_whitelisted"]
+    del os.environ["NOT_PARSED"]
+
```

Comment:
This is good by me, I wonder if we can move this to a decorator for the tests. Something like

```python
@set_environ(IN_TOTO_ARTIFACT_BASE_PATH="thisthing")
def test_this_and_that(self):
  ...
```

It's probably worth doing down the line, but I'd move it to a ticket.

Reply:
Oh I wasn't aware of that decorator. I can do that right away.

Reply:
no, we need to write it

Reply:
oh ok :) then I guess the current approach is fine.